helper/Datatypes.php

```diff
@@ -53,21 +53,38 @@ class Datatypes
      */
     public static function decodeFile($binary)
     {
-        
+
         $returnValue = ['name' => '', 'mime' => '', 'data' => ''];
-        
+
         if (empty($binary) === false) {
             $filenameLen = current(unpack('S', substr($binary, 0, 2)));
+
+            //validate case when unpacked filename length more or equal entire provided string
+            // what mean that provided string definitely in incorrect format
+            if ($filenameLen >= strlen($binary)) {
+                return $returnValue;
+            }
+
             if ($filenameLen > 0) {
                 $returnValue['name'] = substr($binary, 2, $filenameLen);
             }
-            
-            $mimetypeLen = current(unpack('S', substr($binary, 2 + $filenameLen, 2)));
+
+            $packedMimeTypeLen = substr($binary, 2 + $filenameLen, 2);
+            if ($packedMimeTypeLen === false) {
+                return $returnValue;
+            }
+
+            $unpackedMimeTypeLen = unpack('S', $packedMimeTypeLen);
+            if ($unpackedMimeTypeLen === false) {
+                return $returnValue;
+            }
+
+            $mimetypeLen = current($unpackedMimeTypeLen);
             $returnValue['mime'] = substr($binary, 4 + $filenameLen, $mimetypeLen);
```

Comment:
why you add 2 bit?

Reply:
i didn't add any bits, just validation. this is old method of file string decoding which format described in phpdoc

Reply:
yeh sorry didn't mentioned that it was before